File: application/src/main/java/de/pbauerochse/worklogviewer/version/GitHubVersion.java
package de.pbauerochse.worklogviewer.version;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;

/**
 * Created by patrick on 01.11.15.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class GitHubVersion {

    private final String url;
    private final String version;
    private final boolean draft;
    private final Date published;

    @JsonCreator
    public GitHubVersion(@JsonProperty("html_url") String url,
                         @JsonProperty("tag_name") String version,
                         @JsonProperty("draft") boolean draft,
                         @JsonProperty("published_at") Date published) {
        this.url = url;
        this.version = version;
        this.draft = draft;
        this.published = published;
    }

    public String getUrl() {
        return url;
    }

    public String getVersion() {
        return version;
    }

    public boolean isDraft() {
        return draft;
    }

    public boolean isRelease() {
        return !isDraft();
    }

    public Date getPublished() {
        return published;
    }

}

File: application/src/main/java/de/pbauerochse/worklogviewer/version/GitHubVersion.kt
package de.pbauerochse.worklogviewer.version

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

@JsonIgnoreProperties(ignoreUnknown = true)
class GitHubVersion @JsonCreator constructor(
    @JsonProperty("html_url") val url: String,
    @JsonProperty("tag_name") val version: String,
    @JsonProperty("draft") val isDraft: Boolean,
    @JsonProperty("published_at") val published: Date
) {

    val isRelease: Boolean
        @JsonIgnore get() = !isDraft

}
